kornia/augmentation/_2d/intensity/random_rain.py

```diff
@@ -0,0 +1,82 @@
+from typing import Any, Dict, Optional, Tuple
+
+import torch
+
+from kornia.augmentation._2d.intensity.base import IntensityAugmentationBase2D
+from kornia.augmentation.random_generator._2d import RainGenerator
+from kornia.core import Tensor
+from kornia.core.check import KORNIA_CHECK
+
+
+class RandomRain(IntensityAugmentationBase2D):
+    r"""Add Random Rain to the image.
+
+    Args:
+        p: probability of applying the transformation.
+        number_of_drops: number of drops per image
+        drop_height: height of the drop in image(same for each drops in one image)
+        drop_width: width of the drop in image(same for each drops in one image)
+    Shape:
+        - Input: :math:`(C, H, W)` or :math:`(B, C, H, W)`, Optional: :math:`(B, 3, 3)`
+        - Output: :math:`(B, C, H, W)`
+
+    Examples:
+        >>> rng = torch.manual_seed(0)
+        >>> input = torch.rand(1, 1, 5, 5)
+        >>> rain = RandomRain(p=1,drop_height=(1,2),drop_width=(1,2),number_of_drops=(1,1))
+        >>> rain(input)
+        tensor([[[[0.4963, 0.7682, 0.0885, 0.1320, 0.3074],
+                  [0.7843, 0.4901, 0.8964, 0.4556, 0.6323],
+                  [0.3489, 0.4017, 0.0223, 0.1689, 0.2939],
+                  [0.5185, 0.6977, 0.8000, 0.1610, 0.2823],
+                  [0.6816, 0.9152, 0.3971, 0.8742, 0.4194]]]])
+    """
+
+    def __init__(
+        self,
+        same_on_batch: bool = False,
+        p: float = 0.5,
+        keepdim: bool = False,
+        number_of_drops: Tuple[int, int] = (1000, 2000),
```

Comment:
```suggestion
        number_of_drops: tuple[int, int] = (1000, 2000),
```
`from __future__ import annotations`

same for the rest if possible /cc @johnnv1